### PR TITLE
Close and reopen stream after bpeek exits

### LINKED ISSUE #182: LSF manager broken in Julia 1.8.1
```julia
julia> addprocs_lsf(3)
Error launching workers
CapturedException(ClusterManagers.LSFException(""), Any[(lsf_bpeek(manager::LSFManager, jobid::SubString{String}, iarray::Int64) at lsf.jl:61, 1), (lsf_launch_and_monitor(manager::LSFManager, launched::Vector{WorkerConfig}, c::Condition, jobid::SubString{String}, iarray::Int64) at lsf.jl:71, 1), (#31 at lsf.jl:98 [inlined], 1), ((::Base.var"#929#934"{ClusterManagers.var"#31#32"{LSFManager, Vector{WorkerConfig}, Condition, SubString{String}}})(r::Base.RefValue{Any}, args::Tuple{Int64}) at asyncmap.jl:100, 1), (macro expansion at asyncmap.jl:234 [inlined], 1), ((::Base.var"#945#946"{Base.var"#929#934"{ClusterManagers.var"#31#32"{LSFManager, Vector{WorkerConfig}, Condition, SubString{String}}}, Channel{Any}, Nothing})() at task.jl:484, 1)])
Int64[]
```

Seems the culprit is that the `readline` [here](https://github.com/JuliaParallel/ClusterManagers.jl/blob/ff73e36e108cfd6810d6689c51696930a7503f63/src/lsf.jl#L19) returns an empty string if there is no new line in the buffer. It did not do this in 1.6.2 (haven't tried versions in between) where it instead blocked until new data was available in the buffer.

Easy fix is to treat empty line the same as "job not started", but then we'll be busylooping when we should be sleeping. 

Maybe this is an issue with julia 1.8.1? 

Here is a reproducer:

```julia
julia> io = Base.BufferStream()
BufferStream(bytes waiting=0, isopen=true)

julia> @async while true
       @show readline(io)
       end

pipeline(`echo hi`, stdout=io) |> run
readline(io) = "hi"
readline(io) = ""
readline(io) = ""
readline(io) = ""
readline(io) = ""
readline(io) = ""
readline(io) = ""
readline(io) = ""
etc..
```

### PR DESCRIPTION
**Pull request by DrChainsaw**
Fixes #182 

Not sure is this is the right way to go about it. Rather than trying to handle a flood of empty strings we just close the stream and open a new one when we think that bpeek has terminated (i.e when bpeek outputs "Not yet started" ).

Not sure what will/should happen when stream is passed over to Distributed if user does not pass -f. Does it know what to do with empty strings or should we close the stream if the process has exited?

### CODE REVIEW

**Comment by DrChainsaw** [comment]
Bump.

I have tested this both with and without `-f` on my system and it works.

One potential problem with the code is if `bpeek -f` for some reason would output an empty string without having terminated. This could cause us to wait forever until it completes. I guess one could try to parse the `bpeek_flags` to understand if it is expected to terminate or not but I'm worried this will create surprising breakages without actually solving anything.

**Review by kescobo — APPROVED** [review]
Sorry for the delay in getting this reviewed. This is a tough package to maintain because most of the maintainers don't have experience with all of the different cluster managers, and because of the lack of testing. I also don't think anyone really feels a sense of ownership, and therefore also don't feel empowered to make decisions.

Given all of that, I'm inclined to just merge this and make a version increment, then see if anyone complains...

**Comment by DrChainsaw** [comment]
Thanks for the spot.

I agree about the maintainability. I'm trying to do my part as I'm using the LSF manager, but it is always good with a second pair of eyes. 

Especially in this case I'm a bit uncertain what is the right way to handle a text stream as I find the [correct behaviour of `BufferStream`](https://github.com/JuliaLang/julia/issues/46869) a bit difficult to reason about. Can you always assume that empty string means the program terminated and if not, what is the correct way to determine whether it did? Maybe `BufferStream` is just the wrong tool here?

Anways, I will merge this tomorrow or so if there are no more comments. The current behaviour is at least better than just not working at all.

**Comment by DrChainsaw** [comment]
Thanks for the pull. I somehow managed to forget about this PR or else I would have merged it myself long ago.

**Pull request merged**
